### ISSUE TITLE
Maximum Execution Timeout in Reports leads to blank page

### ISSUE THREAD

**Issue by connorshea** [salesagility/SuiteCRM#7914]
#### Issue
I'm trying to migrate some reports created with KReports to the core Reports module, and I get a blank screen after the page fails to load (similar to #5073).

#### Expected Behavior
The page will load and display the details from the report.

#### Actual Behavior
I get an error in the logs like this:

```
[23-Sep-2019 21:32:47 UTC] PHP Fatal error:  Maximum execution time of 60 seconds exceeded in /Users/connorshea/code/sites/SuiteCRM/include/database/MysqliManager.php on line 139
```

#### Possible Fix
There are two main things that need to be fixed here:

1. If the user creates a report that's really slow, it shouldn't cause the page to fail rendering. Instead, the report detailview should be loaded, and then the report can be loaded asynchronously via JavaScript. Right now I have to modify the URL to replace DetailView with EditView in order to fix a report that's become broken like this.
2. Reports should be faster, I don't think I'm doing anything particularly complex that'd warrant such a problem showing up.

#### Steps to Reproduce
1. Create a complex Report in the Reports module.
  - Mine has 11 columns, a sorting rule, and is running on a dataset of about 57,000ish Leads.
  - Optionally, you can also decrease the `max_execution_time` in the php.ini to 5 or 10 seconds, to make it much easier to reproduce the problem.
2. Attempt to view the Report.
3. See that the page fails to load and the `php_errors.log` includes an error like `Maximum execution time of 60 seconds exceeded`.
4. See that there's no way to get to the edit screen from this page (you can still get to the edit page from the reports list view, of course).

#### Context
This is making it hard to migrate the KReports reports to the core module.

#### Your Environment
* SuiteCRM Version used: 7.10.19
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox 70
* Environment name and version (e.g. MySQL, PHP 7): MySQL 5.7, PHP 7.2
* Operating System and version (e.g Ubuntu 16.04): macOS Mojave 10.14.6

**Comment by Abuelodelanada** [comment]
@connorshea can you execute an `EXPLAIN` of the query and paste the result?

**Comment by connorshea** [comment]
Unfortunately, because it times out it doesn't log the query :/

**Comment by Abuelodelanada** [comment]
What about executing `SHOW FULL PROCESSLIST; ` in mysql console while the query is still running?

**Comment by connorshea** [comment]
That doesn't seem to get me anything other than 'sleep' for the crm database, maybe it's slow because of something other than the SQL query 🤔

**Comment by LeoZandvliet** [comment]
Are you on shared hosting or own server? The cpu speed on shared hosting caused me much troubles. I opened this ticket a while ago about the same problem with reports #7705

**Comment by pgorod** [comment]
@LeoZandvliet I hadn't seen that other issue you opened yet, but it's definitely worth exploring. It's very likely that some refactoring there could bring huge benefits.

**Comment by connorshea** [comment]
@LeoZandvliet in this case I was having this problem when hosting the server locally on my MacBook Pro w/ a 2.3 GHz Intel Core i5 and 8GB RAM. Everything else runs perfectly fine so I don't think that'd be the problem.

**Comment by holdusback** [comment]
Following this thread, I have a customised AOR_Report.php (mainly to have a grand total of group total's, and display totals in csv.) and I have this issue too.

Tested with original php file and I have the same error.

**Comment by pgorod** [comment]
@holdusback can you work on a solution based on the information in #7705?

**Comment by holdusback** [comment]
Yeah I've saw this thread, If I catch some time I will do it ! For now I need to fix this issue that we talked about (other issue) : https://community.suitecrm.com/t/googlemaps-geocoding-cron-marked-as-executed-but-doesnt-geocode-anything/70341/7
